core/src/test/java/org/apache/calcite/avatica/util/ArrayImplTest.java

```diff
@@ -164,6 +165,267 @@
     }
   }
 
+  /**
+   * The same test as arrayOfStructs(), except we use List instead of ArrayImpl.
+   *
+   * This test deliberately covers the
+   * org.apache.calcite.avatica.util.AbstractCursor.ArrayAccessor.convertValue()
+   * method, which is called from
+   * org.apache.calcite.avatica.util.AbstractCursor.ArrayAccessor.getObject() when
+   * input is a general List instead of org.apache.calcite.avatica.util.ArrayImpl.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void listOfStructs() throws Exception {
+    // Define the struct type we're creating.
+    ColumnMetaData intMetaData = MetaImpl.columnMetaData("MY_INT", 1, int.class, false);
+    ColumnMetaData stringMetaData = MetaImpl.columnMetaData("MY_STRING", 2, String.class, true);
+    StructType structType = ColumnMetaData.struct(Arrays.asList(intMetaData, stringMetaData));
+    // Create some structs.
+    Struct struct1 = new StructImpl(Arrays.<Object>asList(1, "one"));
+    Struct struct2 = new StructImpl(Arrays.<Object>asList(2, "two"));
+    Struct struct3 = new StructImpl(Arrays.<Object>asList(3));
```

Comment:
Nitpick: try to be consistent in the use of `Collections.singletonList` vs `Arrays.asList(x)` overall in the file